core/src/main/java/com/microsoft/applicationinsights/TelemetryConfiguration.java

```diff
@@ -77,6 +80,33 @@ public static TelemetryConfiguration getActive() {
     }
 
     /**
+     * Set {@link TelemetryConfiguration} instance to be 'Active'.
+     * Telemetry configuration must not be set twice explicitly or through calling {@link TelemetryConfiguration#getActive()}.
+     *
+     * @param telemetryConfiguration preconfigured instance of Telemetry Configuration
+     */
+    public static void setActive(TelemetryConfiguration telemetryConfiguration) {
```

Comment:
We found it messy to allow to replace the configuration object. As you mentioned in comments one of the modules may get a hold of an object and will not pick up an update. Would it be better to update properties of an object instead?

With dependency injection we would simply initialize singleton of dependency injection with the value of `telemetryconfiguration.active`

Reply:
That's why I added throw to be sure that configuration is initialized only once and no module is initialized with wrong `TelemetryConfiguration`. Probably configuration could be changed once context is loaded but I'm afraid that some module can get properties from the `TelemetryConfiguration` when those properties are not set yet - like `WebRequestTrackingFilter#registerWebApp` that is executed only once.